paasta_tools/kubernetes_tools.py

```diff
@@ -1322,9 +1322,11 @@ def get_boto_volume(self) -> Optional[V1Volume]:
         if not secret_hash:
             log.warning(f"Expected to find k8s secret {secret_name} for boto_cfg")
             return None
-        secret_name = f"paasta-boto-key-{service_name}"
+        secret_name = limit_size_with_hash(f"paasta-boto-key-{service_name}")
         volume = V1Volume(
-            name=f"secret-boto-key-{service_name}",
+            name=self.get_sanitised_volume_name(
```

Comment:
we could have also created a `get_boto_key_volume_name` function to be consistent with the other usages of `get_sanitised_volume_name`, but this is also fine imo